packages/alfa-style/src/property/contain.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  Keyword,
+  Token,
+  Tuple,
+  type Parser as CSSParser,
+} from "@siteimprove/alfa-css";
+import { Parser } from "@siteimprove/alfa-parser";
+import { Slice } from "@siteimprove/alfa-slice";
+import { Err, Result } from "@siteimprove/alfa-result";
+
+import { Longhand } from "../longhand.js";
+
+const { either, map } = Parser;
+
+type Specified = Specified.Single | Specified.Multiple;
+type Computed = Specified;
+
+namespace Specified {
+  export type Single = Tuple<
+    [Keyword<"none"> | Keyword<"strict"> | Keyword<"content">]
+  >;
+  export type Multiple = Tuple<
+    [
+      Keyword<"size"> | Keyword<"inline-size">,
+      Keyword<"layout">,
+      Keyword<"style">,
+      Keyword<"paint">,
+    ]
+  >;
+}
```

Comment:
**Suggestion:** Looking at the property definition, it seems that the 4 keywords are more like flags than real values.
This means that the typical questions we'll need to ask about it will be "does this `contain` has `size`"? Storing them as a list of keywords will make that rather cumbersome (`style.computed("contain").includes(Keyword("size"))` is a bit hard to read).

It feels like it would be better to have at least the computed value as 4 boolean, either as a tuple, or as an object, or maybe it makes sense to create a type in `alfa-css` for this property. So we'd internally store 4 booleans but could have the correct stringification to respect the specs.
Then querying it would be `style.comnputed("contain").size` which feels more natural. This should also simplify testing the the `toJSON` would always return the 4 booleans, so order won't matter.